src/client/cmd/launch.cpp

```diff
@@ -292,7 +292,8 @@ mp::ReturnCode cmd::Launch::request_launch()
             {LaunchProgress_ProgressTypes_KERNEL, "Retrieving kernel image: "},
             {LaunchProgress_ProgressTypes_INITRD, "Retrieving initrd image: "},
             {LaunchProgress_ProgressTypes_EXTRACT, "Extracting image: "},
-            {LaunchProgress_ProgressTypes_VERIFY, "Verifying image: "}};
+            {LaunchProgress_ProgressTypes_VERIFY, "Verifying image: "},
+            {LaunchProgress_ProgressTypes_WAITING, "Waiting for in progress image preparation"}};
```

Comment:
I don't think it requires a hyphen as 'in progress' is perfectly acceptable for meaning "taking place" or "under way".  See https://www.collinsdictionary.com/us/dictionary/english/in-progress for how they spell it.

I agree about the colon- I just missed putting that in there.

I'd prefer to be more specific to let the user know that there is already another process of downloading and preparing the image taking place.  It's not necessarily from another client since, at some point, the automatic image updater could be doing this or the pet-env stuff (although neither of those do it right now).

Since I want to be specific, I'm not sure how `Waiting for in progress image preparation:` is less readable than "Image preparation in progress; waiting: ".  My way only adds the word "for", but your way adds a semicolon, so meh.  And my way keeps the consistency of beginning each phase with the present participle verb.

Reply:
Alright, good point, I am convinced on going with your version... except the hyphen :stuck_out_tongue: See https://getitwriteonline.com/articles/hyphenated-adjectives/ or https://www.grammarbook.com/punctuation/hyphens.asp . Saying that "Foo is in progress" is of course correct, but calling it an "in-progress Foo" turns it into a compound adjective and requires an hyphen to be perfectly correct. Just a tiny pedantic detail though...

Reply:
lol, ok, point taken and learned something new:)